pandas/tests/test_downstream.py

```diff
@@ -53,6 +55,11 @@ def test_xarray(df):
     assert df.to_xarray() is not None
 
 
+def test_oo_optimizable():
+    # GH 21071
+    subprocess.check_call(["python", "-OO", "-c", "import pandas"])
```

Comment:
Does pandas import all modules at `import pandas`? If not, maybe [this](https://github.com/bokeh/bokeh/blob/73f60d791310f4a03e9a8c06eb743de496cc2ada/tests/test_python_execution_with_OO.py) will be helpful.

Reply:
@WillAyd do we want to go this way to also check some submodules?

Reply:
I can take a look - if it's not a lot of effort I suppose it doesn't hurt

Reply:
I've been messing around with this for a bit but AFAICT it keeps getting choked up when trying to import modules containing optional dependencies (ex: `pandas.io.s3`). I can't think of a simple way to configure the test so that it can differentiate intentional ImportErrors as a result of missing dependencies.

For now I'm going to revert to just working with the top level import - can open a separate issue if we wanted this to work recursively

Reply:
Yep, if it turns out difficult, on problem in starting with the basic test (which already tests the large majority of our API)